packages/firebase_messaging/firebase_messaging/macos/firebase_messaging/Package.swift

```diff
@@ -0,0 +1,122 @@
+    messagingDirectory,
+    "..",
+    "..",
+    "pubspec.yaml",
+  ])
+  do {
+    let yamlString = try String(contentsOfFile: pubspecPath, encoding: .utf8)
+    let lines = yamlString.split(separator: "\n")
+
+    guard let packageVersionLine = lines.first(where: { $0.starts(with: "version:") }) else {
+      throw ConfigurationError.invalidFormat("No package version line found in pubspec.yaml")
+    }
+    var packageVersion = packageVersionLine.split(separator: ":")[1]
+      .trimmingCharacters(in: .whitespaces)
+      .replacingOccurrences(of: "+", with: "-")
+    packageVersion = packageVersion.replacingOccurrences(of: "^", with: "")
+
+    guard let firebaseCoreVersionLine = lines.first(where: { $0.contains("firebase_core:") }) else {
+      throw ConfigurationError
+        .invalidFormat("No firebase_core dependency version line found in pubspec.yaml")
+    }
+    var firebaseCoreVersion = firebaseCoreVersionLine.split(separator: ":")[1]
+      .trimmingCharacters(in: .whitespaces)
+    firebaseCoreVersion = firebaseCoreVersion.replacingOccurrences(of: "^", with: "")
+
+    return (packageVersion, firebaseCoreVersion)
+  } catch {
+    throw ConfigurationError.fileNotFound("Error loading or parsing pubspec.yaml: \(error)")
+  }
+}
+
+let library_version: String
+let firebase_sdk_version_string: String
+let firebase_core_version_string: String
+let shared_spm_tag = "-firebase-core-swift"
+
+do {
+  library_version = try loadPubspecVersions().packageVersion
+  firebase_sdk_version_string = try loadFirebaseSDKVersion()
+  firebase_core_version_string = try loadPubspecVersions().firebaseCoreVersion
+} catch {
+  fatalError("Failed to load configuration: \(error)")
+}
+
+guard let firebase_sdk_version = Version(firebase_sdk_version_string) else {
+  fatalError("Invalid Firebase SDK version: \(firebase_sdk_version_string)")
+}
+
+guard let shared_spm_version = Version("\(firebase_core_version_string)\(shared_spm_tag)") else {
+  fatalError("Invalid firebase_core version: \(firebase_core_version_string)\(shared_spm_tag)")
+}
+
+let package = Package(
+  name: "firebase_messaging",
+  platforms: [
+    .macOS("10.15"),
+  ],
+  products: [
+    .library(name: "firebase-messaging", targets: ["firebase_messaging"]),
+  ],
+  dependencies: [
+    .package(url: "https://github.com/firebase/firebase-ios-sdk", from: firebase_sdk_version),
+    .package(url: "https://github.com/firebase/flutterfire", exact: shared_spm_version),
+  ],
+  targets: [
+    .target(
+      name: "firebase_messaging",
+      dependencies: [
+        .product(name: "FirebaseMessaging", package: "firebase-ios-sdk"),
+        // Wrapper dependency
+        .product(name: "firebase-core-shared", package: "flutterfire"),
+      ],
+      resources: [
+        .process("Resources"),
+      ],
+      cSettings: [
+        .headerSearchPath("include"),
+        .define("LIBRARY_VERSION", to: "\"\(library_version)\""),
+        .define("LIBRARY_NAME", to: "\"flutter-fire-rc\""),
```

Comment:
```suggestion
        .define("LIBRARY_NAME", to: "\"flutter-fire-fcm\""),
```

Reply:
I honestly didn't know what this identifier stands for ^^